python/cudf/cudf/core/column/string.py

```diff
@@ -4843,6 +4843,7 @@ def character_ngrams(
         result = self._return_or_inplace(lc, retain_index=True)
 
         if isinstance(result, cudf.Series) and not as_list:
+            result = result[result.list.len() > 0] # before exploding, removes those lists which have 0 length
```

Comment:
Seems this fix could go in the calling code (cuml?). 
Just pass `as_list=True` and do the filter step and explode outside this function.

Reply:
This fix should be included in the cuDF repo due to it being a bug in the `character_ngrams` function itself. sklearn does not have this underlying issue with the `sklearn.feature_extraction.text.CountVectorizer` class that causes NA tokens to be returned on empty strings for example.

Reply:
Ok. Could you add a pytest for this?

Reply:
Sure

Reply:
Any progress on this @Vortexx2 ?